services/worker/src/worker/job_runners/config/parquet_and_info.py

```diff
@@ -575,15 +575,22 @@ def get_parquet_file_and_size(url: str, hf_endpoint: str, hf_token: Optional[str
     return pq.ParquetFile(f), f.size
```

Comment:
the fact that we don't close `f` here also seems to result in a bit bigger memory consumption but not that crucial.

Reply:
changed it here [8dd142a](https://github.com/huggingface/datasets-server/pull/2564/commits/8dd142af6c81a479873c0be358ae7824584963a6)